mqtt_cloud_pubsub_proxy/src/main/java/com/google/cloud/pubsub/proxy/gcloud/GcloudPubsub.java

```diff
@@ -95,8 +95,7 @@
   private static final String HASH_PREFIX = "sha256-";
   private static final String PREFIX = "cps-";
   private static final String ASTERISK_URLENCODE_VALUE = "%2A";
-  // TODO - read project name from config file
-  private static final String CLOUD_PUBSUB_PROJECT_NAME = "iot-cloud-pubsub";
+  private static final String CLOUD_PUBSUB_PROJECT_NAME = System.getenv("GCLOUD_PUBSUB_PROJECT_ID");
```

Comment:
If you want you can copy/paste some code from https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-core/src/main/java/com/google/gcloud/ServiceOptions.java#L194 that tries to auto-detect the project id.